### PR TITLE
Fix: Handle null values in rollback 

### PR DESCRIPTION
**Pull request by mohamedveron**

### CODE REVIEW

File: executor/executor.go [hunk 1]
```diff
@@ -313,15 +313,24 @@ func (executor *Executor) execute(
...
-			memoryRequest := container.Resources.Requests.Memory().Value()
-			cpuLimit := container.Resources.Limits.Cpu().MilliValue()
-			cpuRequest := container.Resources.Requests.Cpu().MilliValue()
-
-			*totalResources.Containers[0].Limits.Memory = memoryLimit / 1024 / 1024
-			*totalResources.Containers[0].Requests.Memory = memoryRequest / 1024 / 1024
-			*totalResources.Containers[0].Limits.CPU = cpuLimit
-			*totalResources.Containers[0].Requests.CPU = cpuRequest
+			// handle if requests and limits is null in rollback DEV-2056"
+			if container.Resources.Limits != nil {
```
Comment (MohamedAdel-magalix) [inline executor/executor.go:317]: Shouldn't we need to handle the case of if it's nil by failing the job?

**Review by MohamedAdel-magalix — CHANGES_REQUESTED** [review]

File: executor/executor.go (hunk 1 above)
Comment (mohamedveron) [reply executor/executor.go:317]: no because at least one value will be in the recommendation

Comment (mohamedveron) [reply executor/executor.go:317]: there is no recommendation without both requests and limits

Comment (amrHassanAbdallah) [reply executor/executor.go:317]: @mohamedveron While indeed there is no model in the resource advisor that will generate empty resources object, but what will happen if the logic in the adv were corrupted and somehow generated that recommendation, everything is possible and the logic should be handling any case.

The key point is it important to handle such a case? what's the impact of the resource obj empty and the agent tried to exec it, would it fail? would it stuck in the job forever? if the answer is no in both cases, then it's not much important to handle it now.

**Pull request merged**